modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CommerceAccountSupplierRole.testcase

```diff
@@ -161,6 +337,1148 @@ definition {
+
+					CommerceOrders.editOrderAddressCP(
+						editedCity = "Test City Edit",
+						editedCountry = "Italy",
+						editedName = "Address Name Edit",
+						editedRegion = "Lombardia",
+						editedStreet1 = "Test Address Edit",
+						editedZipCode = "Test Zip Code Edit");
+				}
+
+				for (var termType : list "Payment,Delivery") {
+					CommerceOrders.editOrdersEntryTerms(
+						newTerm = "${termType} Term 1",
+						type = ${termType});
+				}
+
+				Click(
+					key_menuTab = "Payments",
+					locator1 = "CommerceEntry#MENU_TAB");
+
+				CommerceOrders.editPaymentMethod(
+					eligiblePaymentMethods = "Money Order,PayPal",
+					notEligiblePaymentMethods = "Authorize.Net",
+					paymentMethod = "PayPal");
+
+				Click(
+					key_menuTab = "General",
+					locator1 = "CommerceEntry#MENU_TAB");
+
+				CommerceOrders.viewOrdersEntry(
+					addressAddress = "Test Address Edit",
+					addressCity = "Test City Edit",
+					addressZipCode = "Test Zip Code Edit",
+					deliveryTerm = "Delivery Term 1",
+					orderType = "Order Type 1",
+					paymentMethod = "PayPal",
+					paymentTerm = "Payment Term 1",
+					productDiscount = "0.00",
+					productListPrice = "24.00",
+					productName = "U-Joint",
+					productQuantity = 1,
+					productSku = "MIN55861",
+					productTotalPrice = "24.00");
+			}
+
+			task ("And the supplier user can update the order items in the order Items tab") {
+				Click(
+					key_menuTab = "General",
+					locator1 = "CommerceEntry#MENU_TAB");
+
+				CommerceOrders.editOrderItemDetails(
+					orderItemDiscount = "10.00",
+					orderItemPrice = "30.00",
+					orderItemQuantity = 2,
+					orderItemSku = "MIN55861",
+					orderItemTotal = "30.00");
+
+				CommerceOrders.viewOrdersEntry(
+					productDiscount = "10.00",
+					productListPrice = "30.00",
+					productName = "U-Joint",
+					productQuantity = "2.00",
+					productSku = "MIN55861",
+					productTotalPrice = "30.00");
+			}
+
+			task ("And the supplier user can update the order items in the Order Summary tab") {
+				CommerceOrders.editOrderSummary(
+					orderSummarySubtotalEditValue = "30.00",
+					orderSummarySubtotalValue = "24.00",
+					orderSummaryTotalEditValue = "40.00",
+					orderSummaryTotalValue = "39.00");
+
+				CommerceOrders.viewOrderSummary(
+					orderSummarySubtotalValue = "30.00",
+					orderSummaryTotalValue = "40.00");
+			}
+		}
+
+		task ("And the supplier user can complete the order and deliver 2 quantities of the product") {
+			CommerceOrders.checkoutFullWorkflow(
+				sku = "MIN55861",
+				warehouse = "Italy");
+
+			CommerceOrders.openOrdersAdmin();
+
+			CommerceNavigator.gotoOrderEntryViaAdmin(status = "Partially Shipped");
+
+			CommerceOrders.checkoutFullWorkflow(
+				sku = "MIN55861",
+				warehouse = "Italy");
+
+			CommerceOrders.openOrdersAdmin();
+
+			AssertElementPresent(
+				key_orderStatus = "Completed",
+				locator1 = "CommerceOrders#ORDER_ELLIPSIS_BY_STATUS");
+		}
+	}
```

Comment:
@jaydawu I'm afraid these tests might be a bit long and could affect the batches on the CI, please keep them monitored. We can have a discussion about this when we have more information, thanks!